proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+   > Microsoft.CSharp                                                                   4.0.1
+   > Microsoft.NETCore.Platforms                                                        1.1.0
+   > Microsoft.NETCore.Targets                                                          1.1.0
+...
+
+```
+
+### dotnet list package --include-transitive --json
+
+```json
+{
+  "version": 1,
+  "MyProjectA": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "Microsoft.Extensions.Primitives",
+          "requestedVersion": "[1.0.0, 5.0.0]",
+          "resolvedVersion": "1.0.0"
+        },
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "[1.1.2, 2.0.0)",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    }
+  ],
+  "MyProjectB": [
+    {
+      "framework": "netcoreapp3.1",
+      "topLevelPackages": [
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    },
+    {
+      "framework": "net5.0",
+      "topLevelPackages": [ 
+        {
+          "id": "NuGet.Commands",
+          "requestedVersion": "4.8.0-preview3.5278",
+          "resolvedVersion": "4.8.0-preview3.5278"
+        },
+        {
+          "id": "Text2Xml.Lib",
+          "requestedVersion": "1.1.2",
+          "resolvedVersion": "1.1.2"
+        }
+      ],
+      "transitivePackages": [
+        {
+          "id": "Microsoft.CSharp",
+          "resolvedVersion": "4.0.1"
+        },
+        {
+          "id": "Microsoft.NETCore.Platforms",
+          "resolvedVersion": "1.1.0"
+        },
+...
+      ]
+    }
+  ]
+}
+```
+
+### --flatten option
+
+Ability to use new `--flatten` option conjunction with `--json` option for all `dotnet list package` commands to ensure JSON-formatted output is emitted to the console for multiproject solution(including multitarget projects)and all projects and target frameworks are flattened. Same package with multiple version entry is possible. It's useful more for SBOM than single `--json` option.
```

Comment:
This feels like a YAGNI thing. If we can define when this will be used, we can consider it, but I'd say we should just keep the spec as simple as possible with just `--json` for now.

Reply:
Sure, removed.